extmod/modwebsocket.c

```diff
@@ -229,6 +257,34 @@ STATIC mp_uint_t websocket_write(mp_obj_t self_in, const void *buf, mp_uint_t si
         header[3] = size & 0xff;
         hdr_sz = 4;
     }
+    if (self->do_write_masking != NO_WRITE_MASKING) {
+        hdr_sz += 4;
+        header[1] |= 0x80;
+        if (self->do_write_masking == NORMAL_WRITE_MASKING) {
+
+            // RFC6455 Section 5.3 states that the masking key must be derived
+            // from a strong source of entropy.  The "urandom" module doesn't
+            // qualify in this regard, but there isn't any cross-platform
+            // alternative.  Fortunately, the purpose of masking is not
+            // cryptographically motivated.  The "urandom" module should be
+            // seeded though, otherwise upon restart, the same sequence of
+            // masks will always be used.  A seed could be derived from a
+            // network resource, a network interface's characteristics or
+            // statistics, or a platform specific resource.  Examples of using
+            // a platform specific resource include reading an ESP8266's
```

Comment:
I'm all for detailed comments, but specifics of a particular chip in a generic module are superfluous.